.eslintrc.js

```diff
@@ -6,38 +6,138 @@ module.exports = {
+            extends: ['airbnb-typescript', 'plugin:@typescript-eslint/recommended', 'plugin:@typescript-eslint/recommended-requiring-type-checking', 'prettier'],
+            plugins: ['@typescript-eslint', 'jsdoc'],
+            parser: '@typescript-eslint/parser',
+            parserOptions: {
+                project: './tsconfig.json',
+            },
+            rules: {
+                // TODO: Remove the following rules after TypeScript migration is complete.
+                '@typescript-eslint/no-unsafe-argument': 'off',
+                '@typescript-eslint/no-unsafe-call': 'off',
+                '@typescript-eslint/no-unsafe-member-access': 'off',
+                '@typescript-eslint/no-unsafe-assignment': 'off',
+
+                '@typescript-eslint/naming-convention': [
+                    'error',
                     {
-                        name: 'react-native',
-                        importNames: ['useWindowDimensions'],
-                        message: 'Please use useWindowDimensions from src/hooks/useWindowDimensions instead',
+                        selector: 'variable',
+                        format: ['camelCase', 'UPPER_CASE', 'PascalCase'],
                     },
                     {
-                        name: 'react-native',
-                        importNames: ['TouchableOpacity', 'TouchableWithoutFeedback', 'TouchableNativeFeedback', 'TouchableHighlight'],
-                        message: 'Please use PressableWithFeedback and/or PressableWithoutFeedback from src/components/Pressable instead',
+                        selector: ['property', 'function'],
+                        format: ['camelCase', 'PascalCase'],
                     },
                     {
-                        name: 'react-native',
-                        importNames: ['StatusBar'],
-                        message: 'Please use StatusBar from src/libs/StatusBar instead',
+                        selector: ['enumMember', 'typeLike', 'typeParameter'],
+                        format: ['PascalCase'],
+                    },
+                    {
+                        selector: ['parameter', 'method'],
+                        format: ['camelCase'],
+                    },
+                ],
+                '@typescript-eslint/array-type': ['error', {default: 'array-simple'}],
+                '@typescript-eslint/no-unsafe-enum-comparison': 'error',
+                '@typescript-eslint/prefer-enum-initializers': 'error',
+                '@typescript-eslint/no-var-requires': 'off',
+                '@typescript-eslint/no-non-null-assertion': 'error',
+                '@typescript-eslint/no-explicit-any': 'error',
+                '@typescript-eslint/switch-exhaustiveness-check': 'error',
+                '@typescript-eslint/consistent-type-definitions': ['error', 'type'],
+                'react/default-props-match-prop-types': 'off',
+                'react/require-default-props': 'off',
```

Comment:
@blazejkustra @hayata-suenaga I think we can remove these two rules are we won't work with `defaultProps` anymore.

Reply:
Agree, I'll remove it 👌 

From TS config doc:

- 'react/default-props-match-prop-types': 'off',
Disabled because it’s conflicting with TS files and we are not going to have propTypes anymore.
- 'react/require-default-props': 'off',
Disabled because it’s conflicting with TS files and we are not going to have propTypes anymore.